apps/src/storage/dataBrowser/DataLibraryPane.jsx

```diff
@@ -52,17 +59,52 @@ class DataLibraryPane extends React.Component {
       FirebaseStorage.copyStaticTable(datasetInfo.name, () => {}, this.onError);
     }
   };
+  search = e => {
+    let searchValue = '';
+    if (e !== undefined) {
+      searchValue = e.target.value.toLowerCase();
+    }
+    this.setState({
+      search: searchValue
+    });
+  };
+  filterCategories = allCategories => {
+    const searchRegExp = new RegExp('(?:\\s+|^)' + this.state.search, 'i');
+    let potentialCategories = allCategories.map(category => {
```

Comment:
if we use [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) instead of map here, we can map and filter at the same time (instead of having to filter `potentialCategories` on line 86)